### ISSUE TITLE
Vectors as labels

### ISSUE THREAD

**Issue by RawanMG** [BVLC/caffe#2047]
Hi, 
I'm using Caffe for regression and I need to set the labels to be an N-dim vector instead of using a discrete set of labels. Is there a way of doing this without modifying the source code? 

I understand that HDF5 Layer can be used to input that N-dim label vector, but I'm not so sure about vectorizing the image in that case.

**Comment by bhack** [comment]
See also this https://github.com/BVLC/caffe/pull/1522 and cited indirection layer. But probably this code it is far from the HEAD.

**Comment by AlexGruzdev** [comment]
Hi, RawanMG. Currently I'm working on multivariate regression problem and I found that caffe does not support multivariate labels. Moreover, all data layers are designed to store label as integer. The possible solution to your problem is to create your own data layer or store training data directly to blobs.

**Comment by ducha-aiki** [comment]
Use HDF5 input layer

**Comment by shelhamer** [comment]
No, most Caffe layers support matrix and vector data and losses such as
Softmax, Euclidean, and SigmoidCrossEntropy all allow for vector or matrix
ground truth. Caffe networks can have multiple data layers. See this code
example for preparing vector / matrix data:
https://github.com/BVLC/caffe/issues/1698 or use HDF5 data layer as
suggested by Dmytro Mishkin.

An example of multivariate regression / multi-label classification will be
bundled soon to make this more clear.

**Comment by shelhamer** [comment]
For modeling questions ask on the [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) group.
